test/run_test.py

```diff
@@ -862,14 +862,21 @@ def get_selected_tests(options):
             return selected_tests
 
         # Download previous test times to make sharding decisions
-        test_file_times = get_test_times(str(REPO_ROOT), filename=TEST_TIMES_FILE)
-        if len(test_file_times) == 0:
+        path = os.path.join(str(REPO_ROOT), TEST_TIMES_FILE)
+        if os.path.exists(path):
+            with open(path, "r") as f:
+                test_file_times = cast(Dict[str, Any], json.load(f))
+        else:
+            test_file_times = {}
```

Comment:
If the path doesn't exist, would it make sense to fall back to the previous behavior `test_file_times = get_test_times(str(REPO_ROOT), filename=TEST_TIMES_FILE)` here instead?

Reply:
if we do that, we run into the issue of race conditions + reruns being different

Reply:
Oh, what I'm curious about is that may be the possibility of race conditions + reruns being different is still better than the default sharding plan. Thus, we still use the old code knowingly before your fix. If I get your answer correctly, I guess it's not worth the trouble to keep the old logic here as a fallback.